### PR TITLE
[cuDNN][Convolution] Disable cuDNN Fwd Engine 5 in backend version 9.23.1

### LINKED ISSUE #188288: inductor / test-osdc (timm, torchbench) a10g jobs red since cuDNN 9.23.1 upgrade (#187091) — eager non-determinism on conv/BatchNorm models
## Summary

The `inductor / inductor-test / test-osdc (... mt-l-x86aavx2-29-113-a10g)` accuracy
jobs have been **persistently red on `main` since 2026-06-13**, immediately after
the cuDNN wheel upgrade to **9.23.1.3** landed in #187091
(commit ca94d752ae5b0856b7e1538f3e61e7cc501fedda, merged 2026-06-12 20:38 UTC).

The failures are **not** an inductor/compile regression. They are
`eager_two_runs_differ` / `EAGER_FAILED` accuracy failures: the **eager baseline
itself is non-deterministic** across two runs, so the accuracy harness rejects the
model before inductor is even compared. All affected models are
convolution + BatchNorm networks running on sm_86 (A10G), which is the classic
signature of a cuDNN version change.

## Affected jobs (per-commit, `inductor.yml`, trunk)

- `inductor-test / test-osdc (inductor_timm, 1, 2, mt-l-x86aavx2-29-113-a10g)`
- `inductor-test / test-osdc (inductor_timm, 2, 2, mt-l-x86aavx2-29-113-a10g)`
- `inductor-test / test-osdc (inductor_torchbench, 2, 2, mt-l-x86aavx2-29-113-a10g)`
- (also red: `inductor-test / test-osdc (inductor, 1, 2 / 2, 2, ...a10g)`)

## Affected models (status regressed: `eager_two_runs_differ`)

| Suite | Models |
|-------|--------|
| timm | `mobilenetv2_100`, `tf_efficientnet_b0` |
| torchbench | `mobilenet_v2`, `mobilenet_v2_quantized_qat`, `mnasnet1_0`, `shufflenet_v2_x1_0`, `resnet50_quantized_qat`, `opacus_cifar10`, `sam`, `llava` |

Example log excerpt (timm shard 1):
```
mobilenetv2_100   FAIL:  accuracy=eager_two_runs_differ, expected=pass
E torch/_dynamo/utils.py:3728] Accuracy failed: allclose not within tol=0
E torch/_dynamo/utils.py:3544] Accuracy failed for key name blocks.1.0.bn1.bias.grad
Error: 1 models have accuracy status regressed:
mobilenetv2_100   EAGER_FAILED
```

## Evidence of causation

- **Build config on the red commits reports `CUDNN_VERSION=9.23.1`** (CUDA 13.0).
- **Green → red edge lines up with the merge:** inductor runs on 2026-06-10/11
  (pre-merge) were green; every run from 2026-06-13 onward is red with this
  signature. First confirmed red: run 27456681005 (commit a6d2be2c, 2026-06-13),
  identical `mobilenetv2_100 ... eager_two_runs_differ`.
- Failure has been **stable and unchanged for ~2 weeks** (still red as of run
  28250441584, 2026-06-26).
- Failing gradient key (`...bn1.bias.grad`) and the exclusively conv/BatchNorm
  model set point at cuDNN BatchNorm/convolution non-determinism, not at
  torch.compile.

## Suggested next steps

- @eqy / cuDNN folks: confirm whether cuDNN 9.23.1.3 changed default
  BatchNorm/conv algorithm selection on sm_86 to a non-deterministic kernel,
  or whether the accuracy harness needs to pin a deterministic algo for the
  eager baseline.
- Short term, to unblock trunk: either bump the accuracy tolerance / mark these
  models as expected-non-deterministic on A10G in the dynamo benchmark expected
  CSVs, or revert/pin the cuDNN version on this config.

cc @eqy @malfet @nWEIdia @msaroufim (from #187091)

### PR DESCRIPTION
**Pull request by eqy**
to fix https://github.com/pytorch/pytorch/issues/188288, tested that we no longer see nondeterminism on a similar SM 8.6 GPU to A10G (A2)

coached codex to debug and author this

cc @csarofeen @ptrblck @nWEIdia @msaroufim @tinglvv @mruberry @kurtamohler

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

**Comment by IvanKobzarev** [comment]
@eqy 
FYI This PR regressed torch.compiled timm_models runtime +57%, picking conv kernels that are 1.5-2x times slower for those shapes.

```
python benchmarks/dynamo/timm_models.py --performance --inference  --bfloat16 --backend inductor --device cuda
```
